### DISCUSSION TITLE
Run Check failing for "Hello world" example

### DISCUSSION THREAD

**Discussion by cwses1** [concourse/concourse#7331] [Help & Support]
The "Hello World pipeline" (https://concourse-ci.org/hello-world-example.html) example initially failed due to a permissions error.  I have manually configured Postgres, the web node, and a worker node on each of their own Debian Linux (Buster) instances.  This job initially failed with the error message:

`run check: find or create container on worker gator: runc run: exit status 1: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:58: preparing rootfs caused: permission denied
`

I found this strange because the Concourse worker is running as root.  The worker data directory is configured as follows:

`CONCOURSE_WORK_DIR=/var/concourse/worker`

You will notice that the root user owns this directory and has complete permissions:

```
root@gator:/var# ls -la
drwx------  3 root root  4096 Jul 24 12:47 concourse
```

But when I open up the permissions to group and others by running command

`root@gator:/etc# chmod -R 777 /var/concourse`

and I re-run the build, then everything is successful.  This appears to be the only permissions scheme that permits the job to run successfully.  What is the least-privilege permissions mask for the worker data directory, and why would permissions be denied to a process that is running as root?

**Comment by taylorsilva** [comment]
It's not immediately obvious to me why you had to open up permissions so much. As you said, running as root should be more than enough. I haven't done a full non-container install of Concourse; I always use docker or k8s. I haven't seen anyone else report this issue and there are definitely other users that do non-container installs of Concourse. 

The error is coming from runc here: https://github.com/opencontainers/runc/blob/713748d4a8ff418bc8218ffb5f6c80ed396b8c1a/libcontainer/rootfs_linux.go#L775-L792
It's making three `mount` calls, not sure which one could be failing.
